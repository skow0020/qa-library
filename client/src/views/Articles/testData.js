export const articles = {
  data: [
    {
      '_id': '5e220996d7ca7340bcfc9dd5',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Benny Schaden Jr.',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Est quibusdam laboriosam ipsum cum est sunt numquam voluptas. Quas cupiditate facere vero.',
      'article_id': 1000
    },
    {
      '_id': '5e220996d7ca7340bcfc9dd6',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Joelle Mayer',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Dolorum consequuntur suscipit impedit impedit repudiandae qui cupiditate quis rerum. Veritatis ab ab culpa praesentium necessitatibus vitae quasi.',
      'article_id': 1001
    },
    {
      '_id': '5e220996d7ca7340bcfc9dd7',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Leonardo Wunsch',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Quas eius non iure. Voluptatibus cupiditate recusandae libero repellendus velit ex.',
      'article_id': 1002
    }
  ]
};