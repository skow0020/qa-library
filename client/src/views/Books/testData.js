export const books = [
  {
    "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
    "author": "Nova Goldner",
    "category": "API Automation",
    "language": "Ruby",
    "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
    "title": "How to sand a hippo",
    "pdf": "http://www.images.com/crazy.pdf",
    "body": "Est dolor ad. Provident deserunt aliquid sed provident minus fugit.",
    "book_id": 2002
  },
  {
    "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
    "author": "Erin Medhurst",
    "category": "UI Automation",
    "language": "Swift",
    "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
    "title": "How to sand a hippo",
    "pdf": "http://www.images.com/crazy.pdf",
    "body": "Non mollitia dolores dolorum eos unde. Aut enim dolore nesciunt nemo tenetur et rem.",
    "book_id": 1001
  }
];