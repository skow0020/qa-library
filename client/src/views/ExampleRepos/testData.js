export const repos = [
  {
    'id': 198064850,
    'node_id': 'MDEwOlJlcG9zaXRvcnkxOTgwNjQ4NTA=',
    'name': 'acme-outreach',
    'full_name': 'skow0020/acme-outreach',
    'private': false,
    'owner': {
      'login': 'skow0020',
      'id': 12362090,
      'node_id': 'MDQ6VXNlcjEyMzYyMDkw',
      'avatar_url': 'https://avatars2.githubusercontent.com/u/12362090?v=4',
      'gravatar_id': '',
      'url': 'https://api.github.com/users/skow0020',
      'html_url': 'https://github.com/skow0020',
      'followers_url': 'https://api.github.com/users/skow0020/followers',
      'following_url': 'https://api.github.com/users/skow0020/following{/other_user}',
      'gists_url': 'https://api.github.com/users/skow0020/gists{/gist_id}',
      'starred_url': 'https://api.github.com/users/skow0020/starred{/owner}{/repo}',
      'subscriptions_url': 'https://api.github.com/users/skow0020/subscriptions',
      'organizations_url': 'https://api.github.com/users/skow0020/orgs',
      'repos_url': 'https://api.github.com/users/skow0020/repos',
      'events_url': 'https://api.github.com/users/skow0020/events{/privacy}',
      'received_events_url': 'https://api.github.com/users/skow0020/received_events',
      'type': 'User',
      'site_admin': false
    },
    'html_url': 'https://github.com/skow0020/acme-outreach',
    'description': null,
    'fork': false,
    'url': 'https://api.github.com/repos/skow0020/acme-outreach',
    'forks_url': 'https://api.github.com/repos/skow0020/acme-outreach/forks',
    'keys_url': 'https://api.github.com/repos/skow0020/acme-outreach/keys{/key_id}',
    'collaborators_url': 'https://api.github.com/repos/skow0020/acme-outreach/collaborators{/collaborator}',
    'teams_url': 'https://api.github.com/repos/skow0020/acme-outreach/teams',
    'hooks_url': 'https://api.github.com/repos/skow0020/acme-outreach/hooks',
    'issue_events_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues/events{/number}',
    'events_url': 'https://api.github.com/repos/skow0020/acme-outreach/events',
    'assignees_url': 'https://api.github.com/repos/skow0020/acme-outreach/assignees{/user}',
    'branches_url': 'https://api.github.com/repos/skow0020/acme-outreach/branches{/branch}',
    'tags_url': 'https://api.github.com/repos/skow0020/acme-outreach/tags',
    'blobs_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/blobs{/sha}',
    'git_tags_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/tags{/sha}',
    'git_refs_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/refs{/sha}',
    'trees_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/trees{/sha}',
    'statuses_url': 'https://api.github.com/repos/skow0020/acme-outreach/statuses/{sha}',
    'languages_url': 'https://api.github.com/repos/skow0020/acme-outreach/languages',
    'stargazers_url': 'https://api.github.com/repos/skow0020/acme-outreach/stargazers',
    'contributors_url': 'https://api.github.com/repos/skow0020/acme-outreach/contributors',
    'subscribers_url': 'https://api.github.com/repos/skow0020/acme-outreach/subscribers',
    'subscription_url': 'https://api.github.com/repos/skow0020/acme-outreach/subscription',
    'commits_url': 'https://api.github.com/repos/skow0020/acme-outreach/commits{/sha}',
    'git_commits_url': 'https://api.github.com/repos/skow0020/acme-outreach/git/commits{/sha}',
    'comments_url': 'https://api.github.com/repos/skow0020/acme-outreach/comments{/number}',
    'issue_comment_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues/comments{/number}',
    'contents_url': 'https://api.github.com/repos/skow0020/acme-outreach/contents/{+path}',
    'compare_url': 'https://api.github.com/repos/skow0020/acme-outreach/compare/{base}...{head}',
    'merges_url': 'https://api.github.com/repos/skow0020/acme-outreach/merges',
    'archive_url': 'https://api.github.com/repos/skow0020/acme-outreach/{archive_format}{/ref}',
    'downloads_url': 'https://api.github.com/repos/skow0020/acme-outreach/downloads',
    'issues_url': 'https://api.github.com/repos/skow0020/acme-outreach/issues{/number}',
    'pulls_url': 'https://api.github.com/repos/skow0020/acme-outreach/pulls{/number}',
    'milestones_url': 'https://api.github.com/repos/skow0020/acme-outreach/milestones{/number}',
    'notifications_url': 'https://api.github.com/repos/skow0020/acme-outreach/notifications{?since,all,participating}',
    'labels_url': 'https://api.github.com/repos/skow0020/acme-outreach/labels{/name}',
    'releases_url': 'https://api.github.com/repos/skow0020/acme-outreach/releases{/id}',
    'deployments_url': 'https://api.github.com/repos/skow0020/acme-outreach/deployments',
    'created_at': '2019-07-21T14:13:49Z',
    'updated_at': '2019-07-21T14:37:03Z',
    'pushed_at': '2019-07-21T14:37:01Z',
    'git_url': 'git://github.com/skow0020/acme-outreach.git',
    'ssh_url': 'git@github.com:skow0020/acme-outreach.git',
    'clone_url': 'https://github.com/skow0020/acme-outreach.git',
    'svn_url': 'https://github.com/skow0020/acme-outreach',
    'homepage': null,
    'size': 164,
    'stargazers_count': 0,
    'watchers_count': 0,
    'language': 'JavaScript',
    'has_issues': true,
    'has_projects': true,
    'has_downloads': true,
    'has_wiki': true,
    'has_pages': false,
    'forks_count': 0,
    'mirror_url': null,
    'archived': false,
    'disabled': false,
    'open_issues_count': 0,
    'license': null,
    'forks': 0,
    'open_issues': 0,
    'watchers': 0,
    'default_branch': 'master'
  },
  {
    'id': 85446048,
    'node_id': 'MDEwOlJlcG9zaXRvcnk4NTQ0NjA0OA==',
    'name': 'Appium-Tutorial',
    'full_name': 'skow0020/Appium-Tutorial',
    'private': false,
    'owner': {
      'login': 'skow0020',
      'id': 12362090,
      'node_id': 'MDQ6VXNlcjEyMzYyMDkw',
      'avatar_url': 'https://avatars2.githubusercontent.com/u/12362090?v=4',
      'gravatar_id': '',
      'url': 'https://api.github.com/users/skow0020',
      'html_url': 'https://github.com/skow0020',
      'followers_url': 'https://api.github.com/users/skow0020/followers',
      'following_url': 'https://api.github.com/users/skow0020/following{/other_user}',
      'gists_url': 'https://api.github.com/users/skow0020/gists{/gist_id}',
      'starred_url': 'https://api.github.com/users/skow0020/starred{/owner}{/repo}',
      'subscriptions_url': 'https://api.github.com/users/skow0020/subscriptions',
      'organizations_url': 'https://api.github.com/users/skow0020/orgs',
      'repos_url': 'https://api.github.com/users/skow0020/repos',
      'events_url': 'https://api.github.com/users/skow0020/events{/privacy}',
      'received_events_url': 'https://api.github.com/users/skow0020/received_events',
      'type': 'User',
      'site_admin': false
    },
    'html_url': 'https://github.com/skow0020/Appium-Tutorial',
    'description': 'Personal project for learning Appium - In Progress',
    'fork': false,
    'url': 'https://api.github.com/repos/skow0020/Appium-Tutorial',
    'forks_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/forks',
    'keys_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/keys{/key_id}',
    'collaborators_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/collaborators{/collaborator}',
    'teams_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/teams',
    'hooks_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/hooks',
    'issue_events_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues/events{/number}',
    'events_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/events',
    'assignees_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/assignees{/user}',
    'branches_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/branches{/branch}',
    'tags_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/tags',
    'blobs_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/blobs{/sha}',
    'git_tags_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/tags{/sha}',
    'git_refs_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/refs{/sha}',
    'trees_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/trees{/sha}',
    'statuses_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/statuses/{sha}',
    'languages_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/languages',
    'stargazers_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/stargazers',
    'contributors_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/contributors',
    'subscribers_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/subscribers',
    'subscription_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/subscription',
    'commits_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/commits{/sha}',
    'git_commits_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/git/commits{/sha}',
    'comments_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/comments{/number}',
    'issue_comment_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues/comments{/number}',
    'contents_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/contents/{+path}',
    'compare_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/compare/{base}...{head}',
    'merges_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/merges',
    'archive_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/{archive_format}{/ref}',
    'downloads_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/downloads',
    'issues_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/issues{/number}',
    'pulls_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/pulls{/number}',
    'milestones_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/milestones{/number}',
    'notifications_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/notifications{?since,all,participating}',
    'labels_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/labels{/name}',
    'releases_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/releases{/id}',
    'deployments_url': 'https://api.github.com/repos/skow0020/Appium-Tutorial/deployments',
    'created_at': '2017-03-19T02:36:41Z',
    'updated_at': '2017-03-20T14:03:03Z',
    'pushed_at': '2017-03-31T01:34:23Z',
    'git_url': 'git://github.com/skow0020/Appium-Tutorial.git',
    'ssh_url': 'git@github.com:skow0020/Appium-Tutorial.git',
    'clone_url': 'https://github.com/skow0020/Appium-Tutorial.git',
    'svn_url': 'https://github.com/skow0020/Appium-Tutorial',
    'homepage': '',
    'size': 126,
    'stargazers_count': 0,
    'watchers_count': 0,
    'language': 'Java',
    'has_issues': true,
    'has_projects': true,
    'has_downloads': true,
    'has_wiki': true,
    'has_pages': false,
    'forks_count': 0,
    'mirror_url': null,
    'archived': false,
    'disabled': false,
    'open_issues_count': 0,
    'license': null,
    'forks': 0,
    'open_issues': 0,
    'watchers': 0,
    'default_branch': 'master'
  }
];
