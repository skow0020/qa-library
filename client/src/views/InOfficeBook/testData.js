export const inOfficeBook = {
  data: [{
    'office_book_id': '1000',
    'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
    'author': 'Nova Goldner',
    'category': 'API Automation',
    'url': 'https://tap-library-dev.herokuapp.com/tap-dashboard',
    'title': 'How to sand a hippo',
    'body': 'Est dolor ad. Provident deserunt aliquid sed provident minus fugit.',
    'totalCopies': 6,
    'copiesCheckedOut': 2,
    'users': ['stanislov342', 'georgio232']
  }]
};