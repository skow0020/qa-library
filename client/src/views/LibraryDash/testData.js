export const inOfficeBooks = {
  data: [
      {
        "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
        "author": "Nova Goldner",
        "category": "API Automation",
        "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
        "title": "How to sand a hippo",
        "body": "Est dolor ad. Provident deserunt aliquid sed provident minus fugit.",
        "totalCopies": 6,
        "copiesCheckedOut": 2,
        "users": ["stanislov342", "georgio232"]
      },
      {
        "backgroundImage": "https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg",
        "author": "Erin Medhurst",
        "category": "UI Automation",
        "url": "https://tap-library-dev.herokuapp.com/tap-dashboard",
        "title": "How to sand a hippo",
        "body": "Non mollitia dolores dolorum eos unde. Aut enim dolore nesciunt nemo tenetur et rem.",
        "totalCopies": 2,
        "copiesCheckedOut": 1,
        "users": ["georgio232"]
      }
    ]
};