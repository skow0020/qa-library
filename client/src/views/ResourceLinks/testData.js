export const resourceLinks = {
  data: [
    {
      '_id': '5e220997d7ca7340bcfc9ded',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Corporis laborum assumenda non qui. Soluta facilis dolore doloremque enim facere a accusantium.',
      'res_id': 1000
    },
    {
      '_id': '5e220997d7ca7340bcfc9dee',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Voluptatem qui dolore veritatis ullam tempore eos. Deserunt autem nulla praesentium magnam dolorem.',
      'res_id': 1001
    },
    {
      '_id': '5e220997d7ca7340bcfc9def',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Illo qui consequatur eum. Tenetur sed vel est quod.',
      'res_id': 1002
    }
  ]
};