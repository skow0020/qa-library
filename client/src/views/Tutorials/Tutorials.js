/* eslint jsx-a11y/anchor-is-valid: 0 */

import {
  Badge,
  Button,
  Card,
  CardBody,
  Col,
  Container,
  Form,
  FormGroup,
  Row
} from "shards-react";

import CategoriesSelection from "../../components/common/CategoriesSelection";
import LanguagesSelection from "../../components/common/LanguagesSelection";
import LoadError from "../../components/common/LoadError";
import Loading from "../../components/common/Loading";
import PageTitle from "../../components/common/PageTitle";
import React from "react";
import { getCategoryTheme } from "../../utils/util";

class Tutorials extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tutorials: [],
      isLoading: false
    };

    this.getTutorials = this.getTutorials.bind(this);
    this.handleFilterChange = this.handleFilterChange.bind(this);
    this.handleLanguageChange = this.handleLanguageChange.bind(this);
  }

  componentDidMount() {
    this.getTutorials();
  }

  getTutorials() {
    const categoryFilter = this.state.category ? `category=${this.state.category}` : '';
    const languageFilter = this.state.language ? `language=${this.state.language}` : '';
    const filter = `${categoryFilter}&${languageFilter}`;

    this.setState({ isLoading: true });
    fetch(`/api/tutorials?${filter}`)
      .then(response => response.json())
      .then(
        data => this.setState({ tutorials: data.data, isLoading: false }),
        error => this.setState({ error, isLoading: false })
      );
  }

  nextPath(path) {
    this.props.history.push(path);
  }

  handleFilterChange(e) {
    this.setState({
      category: e.target.value
    }, () => this.getTutorials());
  }

  handleLanguageChange(e) {
    this.setState({
      language: e.target.value
    }, () => this.getTutorials());
  }

  render() {
    const { tutorials, isLoading, error, category, language } = this.state;

    if (isLoading) return <Loading />;

    if (error) return <LoadError error="Tutorials failed to load" />;

    return (
      <Container fluid className="main-content-container px-3">
        <Row noGutters className="form-inline py-2">
          <PageTitle sm="8" title="Tutorials" className="text-sm-left" />
          <Button id="add-tutorial" type="button" className="btn btn-success btn-lg" onClick={() => this.nextPath('/add-tutorial')}>
            Add Tutorial
          </Button>
        </Row>
        <Row noGutters className="form-inline py-2">
          <Form id='filtering-form'>
            <FormGroup >
              <CategoriesSelection value={category} onChange={this.handleFilterChange} />
              <LanguagesSelection value={language} onChange={this.handleLanguageChange} />
            </FormGroup>
          </Form>
        </Row>
        <Row>
          {tutorials.map((post, idx) => (
            <Col lg="4" md="6" sm="12" className="mb-4" key={idx}>
              <Card small id={`tutorial-card-${idx}`} className="card-post card-post--1">
                <a href={post.url} target="_blank" rel="noopener noreferrer">
                  <div
                    className="card-post__image"
                    style={{ backgroundImage: `url('${post.backgroundImage}')` }}
                  />
                </a>
                <Badge pill className={`card-post__category bg-${getCategoryTheme(post.category)}`}>
                  {post.category}
                </Badge>

                <CardBody>
                  <h5 className="card-title">
                    <a href={post.url} className="text-fiord-blue" target="_blank" rel="noopener noreferrer">
                      {post.title}
                    </a>
                  </h5>
                  <p className="card-text d-inline-block mb-3">{post.body}</p>
                  <span className="text-muted">{post.date}</span>
                </CardBody>
              </Card>
            </Col>
          ))}
        </Row>
      </Container>
    );
  }
}

export default Tutorials;
