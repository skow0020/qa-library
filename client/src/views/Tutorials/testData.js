export const tutorials = {
  data: [
    {
      '_id': '5e220997d7ca7340bcfc9df6',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Itzel Wiza DVM',
      'category': 'General',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Ut nihil optio est est vel assumenda cum totam sunt. Mollitia incidunt nulla aperiam et aut nobis.',
      'tut_id': 1000
    },
    {
      '_id': '5e220997d7ca7340bcfc9df7',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Annalise Fahey',
      'category': 'UI Automation',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Et temporibus animi eius et inventore minus sunt voluptatum est. Eveniet consectetur voluptatem.',
      'tut_id': 1001
    },
    {
      '_id': '5e220997d7ca7340bcfc9df8',
      'backgroundImage': 'https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg',
      'author': 'Shayne Lubowitz',
      'category': 'DeVops',
      'language': 'Swift',
      'url': 'https://qa-library-dev.herokuapp.com/qa-dashboard',
      'title': 'How to sand a hippo',
      'body': 'Veniam labore vel unde est esse. Provident dolores laudantium consequatur.',
      'tut_id': 1002
    }
  ]
};