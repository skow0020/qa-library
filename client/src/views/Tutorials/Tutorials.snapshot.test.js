import React from 'react';
import Tutorials from './Tutorials';
import { shallow } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import { tutorials } from './testData';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('Tutorials Snapshot Tests', () => {
  const state = {
    tutorials: tutorials,
    isLoading: false,
    category: "UI Automation",
    language: "Python"
  };

  test('Tutorials loading snapshot', () => {
    const wrapper = shallow(<Tutorials />);

    wrapper.setState({ isLoading: true });

    expect(wrapper).toMatchSnapshot();
  });

  test('Tutorials list snapshot ', () => {
    const wrapper = shallow(<Tutorials />);

    wrapper.setState(state);

    expect(wrapper).toMatchSnapshot();
  });
});