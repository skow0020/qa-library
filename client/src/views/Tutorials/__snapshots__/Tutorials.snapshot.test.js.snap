// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorials Snapshot Tests Tutorials list snapshot  1`] = `
<Container
  className="main-content-container px-3"
  fluid={true}
  tag="div"
>
  <Row
    className="form-inline py-2"
    noGutters={true}
    tag="div"
  >
    <PageTitle
      className="text-sm-left"
      sm="8"
      title="Tutorials"
    />
    <Button
      className="btn btn-success btn-lg"
      id="add-tutorial"
      onClick={[Function]}
      tag="button"
      theme="primary"
      type="button"
    >
      Add Tutorial
    </Button>
  </Row>
  <Row
    className="form-inline py-2"
    noGutters={true}
    tag="div"
  >
    <Form
      id="filtering-form"
      tag="form"
    >
      <FormGroup
        tag="div"
      >
        <CategoriesSelection
          onChange={[Function]}
          value="UI Automation"
        />
        <LanguagesSelection
          onChange={[Function]}
          value="Python"
        />
      </FormGroup>
    </Form>
  </Row>
  <Row
    tag="div"
  >
    <Col
      breakpoints={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      className="mb-4"
      key="0"
      lg="4"
      md="6"
      sm="12"
      tag="div"
    >
      <Card
        className="card-post card-post--1"
        id="tutorial-card-0"
        small={true}
        tag="div"
      >
        <a
          aria-label="Navigate to the tutorial url"
          href="https://tap-library-dev.herokuapp.com/tap-dashboard"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="card-post__image"
            style={
              Object {
                "backgroundImage": "url('https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg')",
              }
            }
          />
        </a>
        <Badge
          className="card-post__category bg-info"
          outline={false}
          pill={true}
          tag="span"
          theme="primary"
        >
          API Automation
        </Badge>
        <CardBody
          tag="div"
        >
          <h5
            className="card-title"
          >
            <a
              aria-label="Navigate to the tutorial url"
              className="text-fiord-blue"
              href="https://tap-library-dev.herokuapp.com/tap-dashboard"
              rel="noopener noreferrer"
              target="_blank"
            >
              How to sand a hippo
            </a>
          </h5>
          <p
            className="card-text d-inline-block mb-3"
          >
            Est dolor ad. Provident deserunt aliquid sed provident minus fugit.
          </p>
          <span
            className="text-muted"
          />
        </CardBody>
      </Card>
    </Col>
    <Col
      breakpoints={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
      className="mb-4"
      key="1"
      lg="4"
      md="6"
      sm="12"
      tag="div"
    >
      <Card
        className="card-post card-post--1"
        id="tutorial-card-1"
        small={true}
        tag="div"
      >
        <a
          aria-label="Navigate to the tutorial url"
          href="https://tap-library-dev.herokuapp.com/tap-dashboard"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="card-post__image"
            style={
              Object {
                "backgroundImage": "url('https://i-cdn.phonearena.com/images/article/104071-two_lead/Just-for-you-Here-are-10-paid-iOS-apps-free-for-a-limited-time.jpg')",
              }
            }
          />
        </a>
        <Badge
          className="card-post__category bg-warning"
          outline={false}
          pill={true}
          tag="span"
          theme="primary"
        >
          UI Automation
        </Badge>
        <CardBody
          tag="div"
        >
          <h5
            className="card-title"
          >
            <a
              aria-label="Navigate to the tutorial url"
              className="text-fiord-blue"
              href="https://tap-library-dev.herokuapp.com/tap-dashboard"
              rel="noopener noreferrer"
              target="_blank"
            >
              How to sand a hippo
            </a>
          </h5>
          <p
            className="card-text d-inline-block mb-3"
          >
            Non mollitia dolores dolorum eos unde. Aut enim dolore nesciunt nemo tenetur et rem.
          </p>
          <span
            className="text-muted"
          />
        </CardBody>
      </Card>
    </Col>
  </Row>
</Container>
`;

exports[`Tutorials Snapshot Tests Tutorials loading snapshot 1`] = `<Loading />`;
